test/unit/teacher_set_test.rb

```diff
@@ -182,13 +193,293 @@ class TeacherSetTest < ActiveSupport::TestCase
+    it 'test delete_teacherset_record_from_es method' do
+      expected_resp = {"_index" => "teacherset", "_type" => "teacherset", "_id" => @teacher_set.id, "_version" => 2, "result" => "deleted", 
+        "_shards" => {"total" => 2, "successful" => 1, "failed" => 0}, "_seq_no" => 294, "_primary_term" => 2}
+      @mintest_mock1.expect(:delete_document_by_id, expected_resp, [@teacher_set.id])
+      resp = nil
+      ElasticSearch.stub :new, @mintest_mock1 do
+        resp = @teacher_set.delete_teacherset_record_from_es
+      end
+      @mintest_mock1.verify
+      assert_equal(184, resp)
+    end
+
+    it 'test StandardError while deleting elastic-search document' do
+      expected_resp = {"_index" => "teacherset", "_type" => "teacherset", "_id" => @teacher_set.id, "_version" => 2, "result" => "deleted", 
+        "_shards" => {"total" => 2, "successful" => 1, "failed" => 0}, "_seq_no" => 294, "_primary_term" => 2}
+      @mintest_mock1.expect(:delete_document_by_id, expected_resp, [nil])
+      resp = nil
+      resp = assert_raises(StandardError) do
+        ElasticSearch.stub :new, @mintest_mock1 do
+          resp = @teacher_set.delete_teacherset_record_from_es
+        end
+      end
+      assert_equal(ELASTIC_SEARCH_STANDARD_EXCEPTION[:code], resp.code)
+      assert_equal(ELASTIC_SEARCH_STANDARD_EXCEPTION[:msg], resp.message)
+    end
+  end
+
+
+  # test teacher set object method.
+  describe "test teacher set object" do
+    it 'test teacher set object' do
+      resp = @teacher_set.teacher_set_info
+      assert_equal(@teacher_set.title, resp[:title])
+      assert_equal(@teacher_set.bnumber, resp[:bnumber])
+      assert_equal(@teacher_set.subjects.count, resp[:subjects].count)
+    end
+  end
+
+
+  describe 'update set_type value in teacher_set table' do
+    it 'test update_set_type value method' do
+      resp = nil
+      set_type_val = @teacher_set3.set_type
+      @mintest_mock1.expect(:call, set_type_val, [set_type_val])
+      @mintest_mock2.expect(:call, true, [set_type: set_type_val])
+
+      @teacher_set3.stub :derive_set_type, @mintest_mock1 do
+        @teacher_set3.stub :update, @mintest_mock2 do
+          resp = @teacher_set3.update_set_type(set_type_val)
+        end
+      end
+      assert_equal(set_type_val, resp)
+      @mintest_mock1.verify
+      @mintest_mock2.verify
+    end
+
+    it 'test StandardError while updating the set_type' do
+      resp = nil
+      teacher_set = TeacherSet.new
+      set_type = "Topic Set"
+      @mintest_mock1.expect(:call, set_type)
+
+      resp = assert_raises(StandardError) do
+        teacher_set.stub :derive_set_type, @mintest_mock1 do
+          resp = teacher_set.update_set_type(set_type)
+        end
+      end
+      assert_equal(TEACHERSET_SETTYPE_ERROR[:code], resp.code)
+      assert_equal(TEACHERSET_SETTYPE_ERROR[:msg], resp.message)
+    end
+  end
+
+
+  describe 'get set-type value' do
+    it 'test book club set set_type value' do
```

Comment:
I don't think we use `get_set_type` anywhere anymore — should delete the method and this test and the next one. (It is used in one migration, but that should probably be updated to make it a 'no-op' migration.)